### ISSUE TITLE
Mouse-wheel event for scrolling.

### ISSUE THREAD

**Issue by marcwhitbread** [iosscripts/iosslider#58]
Add an option to have the users mouse-wheel move the slider.

**Comment by dwehrmann** [comment]
Would love to see that implemented.

**Comment by nicooprat** [comment]
+1 !

**Comment by iosscripts** [comment]
Possible workaround when jquery-mousewheel library is included:

``` js
$('.iosSlider').bind('mousewheel', function(event, delta) {

    var currentSlide = $('.iosSlider').data('args').currentSlideNumber;

    //if delta is a positive number, go to prev slide. If delta is a negative number, go to next slide.
    if(delta > 0) {

        $('.iosSlider').iosSlider('goToSlide', currentSlide - 1);

    } else {

        $('.iosSlider').iosSlider('goToSlide', currentSlide + 1);

    }

});
```

This does not include conditions for sliding beyond the min/max of the slider so I'm not sure how that will behave.

**Comment by taylorehat** [comment]
+1!

Or need an easy way to update "transform" values, not just the slide number. The above example works great for "carousel or single slide per view" designs, because it imitates a single touch swipe, but what about multiple slides per view? Only being able to adjust the slider one slide at a time is a clunky user experience, not a true fluid scroll.

I was able to recreate the setSliderOffset function to update the transform values passing both "getSliderOffset + delta \* speed" on a mousewheel event, but it became inefficient to rewrite all the checks for max and min scroll.

**Comment by NKjoep** [comment]
uhm... and what about mice with left-right buttons in the wheel? 

Most of the modern mice have that, you can scroll the whell up-down, click the wheel (middle button) and go left-right :)

**Comment by uberstudio** [comment]
A working hack based on iosscripts' previous example which takes slider limits as a consideration.
(Also works for multiple sliders on same page)

<pre>
   $('.iosSlider').bind('mousewheel', function(event, delta) {
       var slider = $(this);
        var currentSlide = slider.data('args').currentSlideNumber;
        if(delta > 0 && currentSlide != 1) {
            slider.iosSlider('goToSlide', currentSlide - 1);
        } else if (0 > delta && currentSlide != slider.data('args').numberOfSlider) {
            slider.iosSlider('goToSlide', currentSlide + 1);
        }
    });
</pre>

**Comment by davidemorotti** [comment]
A working solution (require custom setting "infiniteSlider: false") based on previous comment by @iosscripts:
Required: Mousewheel plugin (https://github.com/brandonaaron/jquery-mousewheel) by Brandon Aaron.

The "mouse wheel" interaction begin from the first slide ending at the last.

```
var count     = 1;
  $('.iosSlider').bind('mousewheel', function(event, delta) {
    var currentSlide = $('.iosSlider').data('args').currentSlideNumber;
    var lastSlide = $('.iosSlider').data('iosslider').numberOfSlides;

    //if delta is a positive number, go to prev slide. If delta is a negative number, go to next slide.
    if(delta > 0 && count > 1) {
      count --;
      $('.iosSlider').iosSlider('goToSlide', count);
    } else if (delta < 0 && count < lastSlide) {
      count ++;
      $('.iosSlider').iosSlider('goToSlide', count);
    }
 });
```

**Comment by rniswonger** [comment]
To get this working on an infinite slider I triggered 'prevSlide' and 'nextSlide' instead of 'goToSlide'.  The animation won't be as smooth as the 'goToSlide' examples, but at least the slider wraps correctly.  Based on the above examples (the Mousewheel plugin is required):

<pre>
  $('.iosSlider').bind('mousewheel', function(event, delta) {
    var slider = $(this);
    if(delta &gt; 0) {
      slider.iosSlider('prevSlide');
    } else if (delta &lt; 0) {
      slider.iosSlider('nextSlide');
    }
  });
</pre>

**Comment by marcwhitbread** [comment]
@rniswonger you're right. When I originally wrote this the prevSlide and nextSlide public methods didn't exist yet! 

Definitely a better solution. :)
